cloud-aws-common/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/common/connector/resource/AwsCloudWatchCommonServiceTest.java

```diff
@@ -0,0 +1,117 @@
+import com.sequenceiq.cloudbreak.cloud.aws.common.client.AmazonCloudWatchClient;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+
+@ExtendWith(MockitoExtension.class)
+public class AwsCloudWatchCommonServiceTest {
+
+    private static final String REGION = "region";
+
+    private static final String BUCKET_NAME = "bucket";
+
+    @Mock
+    private CommonAwsClient awsClient;
+
+    @Mock
+    private CloudCredential cloudCredential;
+
+    @Mock
+    private AmazonCloudWatchClient amazonCloudWatchClient;
+
+    @Mock
+    private GetMetricStatisticsResult result;
+
+    @InjectMocks
+    private AwsCloudWatchCommonService underTest;
+
+    @Captor
+    private ArgumentCaptor<GetMetricStatisticsRequest> requestCaptor;
+
+    private Date startTime;
+
+    private Date endTime;
+
+    @BeforeEach
+    public void setUp() {
+        startTime = Date.from(Instant.now().minus(42, ChronoUnit.MINUTES));
+        endTime = Date.from(Instant.now());
+    }
+
+    @Test
+    public void getBucketSizeCloudWatchClientThrowsAmazonCloudWatchException() {
+        when(awsClient.createCloudWatchClient(any(), eq(REGION))).thenReturn(amazonCloudWatchClient);
+        AmazonCloudWatchException cwException = new AmazonCloudWatchException("CW error");
+        when(amazonCloudWatchClient.getMetricStatisticsResult(any())).thenThrow(cwException);
+
+        CloudConnectorException ex = assertThrows(CloudConnectorException.class,
+                () -> underTest.getBucketSize(cloudCredential, REGION, startTime, endTime, BUCKET_NAME));
+
+        assertEquals(String.format("Cannot get BucketSizeBytes for bucket %s and timeframe from %s to %s. Reason: %s",
+                BUCKET_NAME, startTime, endTime, cwException.getMessage()), ex.getMessage());
+        assertEquals(cwException, ex.getCause());
+    }
+
+    @Test
+    public void getBucketSizeAwsClientThrowsOtherException() {
+        when(awsClient.createCloudWatchClient(any(), eq(REGION))).thenThrow(new RuntimeException("Other error"));
+
+        Exception ex = assertThrows(Exception.class, () -> underTest.getBucketSize(cloudCredential, REGION, startTime, endTime, BUCKET_NAME));
+
+        assertEquals("Other error", ex.getMessage());
+    }
+
+    @Test
+    public void getBucketSizeWorksCorrectly() {
+        when(awsClient.createCloudWatchClient(any(), eq(REGION))).thenReturn(amazonCloudWatchClient);
+        when(amazonCloudWatchClient.getMetricStatisticsResult(any())).thenReturn(result);
+
+        GetMetricStatisticsResult returnedResult = underTest.getBucketSize(cloudCredential, REGION, startTime, endTime, BUCKET_NAME);
+
+        assertEquals(result, returnedResult);
+        verify(amazonCloudWatchClient, times(1)).getMetricStatisticsResult(requestCaptor.capture());
+
+        GetMetricStatisticsRequest request = requestCaptor.getAllValues().stream().findFirst().get();
```

Comment:
for the future: `requestCaptor.getValue()`

Reply:
Good catch, thanks. I used it in the other tests and forgot to change it here but fixed it now.